pkg/virt-launcher/virtwrap/postcopy.go

```diff
@@ -0,0 +1,66 @@
+package virtwrap
+
+import (
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	cmdclient "kubevirt.io/kubevirt/pkg/virt-handler/cmd-client"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
+
+	libvirt "libvirt.org/libvirt-go"
+)
+
+func (l *LibvirtDomainManager) registerIterationEventForPostCopy(vmi *v1.VirtualMachineInstance, dom cli.VirDomain, options *cmdclient.MigrationOptions, migrationError chan error) {
+	l.virConn.DomainEventMigrationIterationRegister(func(c *libvirt.Connect, d *libvirt.Domain, event *libvirt.DomainEventMigrationIteration) {
+		if vmiHasLocalStorage(vmi) {
```

Comment:
> Do we consider a pvc which can be backed by local storage?

I did not, I can add that to the check 

>Can this be evaluated in the creation time of VM/VMI?(This would give better feedback to user...)

Would the expected UX be the VM/VMI can not be created if this is the case? What if the system level migration strategy is set to postcopy? can vm/vmis with local storage not be created at all?

Reply:
I am not sure if there is a way to check if pvc local...

Yes, I thought that we could not allow it or fire an alert(if we can't be sure)... So there are two situations:
1. Cluster-wide setting is postcopy and VMI is not specified/set to postcopy 
2. Cluster-wide setting is precopy but VMI override this to postcopy

In both, I would be inclined to not allow it as the system can't satisfy it.

Reply:
Currently, virt-handler decides whether VMI is `migratable` during the VMI start. When migration spec is posted, we can quickly reject it for non-migratable VMIs.
Please take a look at [1]

However, I think that `vmiHasLocalStorage` may not be necessary as I mentioned in another comment - but maybe I'm missing something. Please correct me if I'm wrong here :)

[1] https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L728